ql/src/java/org/apache/hadoop/hive/ql/optimizer/SharedWorkOptimizer.java

```diff
@@ -1646,7 +1646,7 @@ private static Set<Operator<?>> gatherDPPBranchOps(ParseContext pctx,
   }
 
   private static void removeBranch(Operator<?> currentOp, Set<Operator<?>> branchesOps,
-          Set<Operator<?>> discardableOps) {
+          SharedWorkOptimizerCache optimizerCache, Set<Operator<?>> discardableOps) {
```

Comment:
could we move the cache to the last param?

Reply:
Thank you @deniskuzZ , I will address this.